src/Host.cpp

```diff
@@ -683,49 +819,345 @@ void Host::send(QString cmd, bool wantPrint, bool dontExpandAliases)
     }
 }
 
-int Host::createStopWatch()
+QPair<int, QString> Host::createStopWatch(const QString& name)
 {
-    int newWatchID = mStopWatchMap.size() + 1;
-    mStopWatchMap[newWatchID] = QTime(0, 0, 0, 0);
-    return newWatchID;
+    if (mResetProfile || mIsProfileLoadingSequence) {
+        // Don't create stopwatches when test loading scripts or during a profile reset:
+        return qMakePair(0, QStringLiteral("unable to create a stopwatch at this time"));
+    }
+
+    if (!mStopWatchMap.isEmpty() && !name.isEmpty()) {
+        // As we need to check the names we will need a different algorithm
+        // than just looking for the first gap in the integer sequence
+        // 1,2,3,...n if we want to avoid going through the list twice,
+        // the first time to check the name is not in use and a second time
+        // to find the lower positive unused integer id:
+        QMapIterator<int, stopWatch*> itStopWatch(mStopWatchMap);
+        while (itStopWatch.hasNext()) {
+            itStopWatch.next();
+            if (itStopWatch.value()->name() == name) {
+                return qMakePair(0, QStringLiteral("a stopwatch with id %1 called \"%2\" already exists").arg(QString::number(itStopWatch.key()), name));
+            }
+        }
+    }
+    int newWatchId = 1;
+    while (mStopWatchMap.contains(newWatchId)) {
+        ++newWatchId;
+    };
+
+    // It is hard to imagine a situation in which this will fail - so we won't
+    // bother coding for it:
+    auto pStopWatch = new stopWatch();
+    Q_ASSERT_X(pStopWatch, "Host::createStopWatch", "out of memory, unable to create new stopwatch");
+    if (!name.isEmpty()) {
+        pStopWatch->setName(name);
+    }
+
+    mStopWatchMap.insert(newWatchId, pStopWatch);
+    return qMakePair(newWatchId, QString());
 }
 
-double Host::getStopWatchTime(int watchID)
+bool Host::destroyStopWatch(const int id)
 {
-    if (mStopWatchMap.contains(watchID)) {
-        return static_cast<double>(mStopWatchMap[watchID].elapsed()) / 1000;
-    } else {
-        return -1.0;
+    auto pStopWatch = mStopWatchMap.take(id);
+    if (!pStopWatch) {
+        return false;
     }
+
+    delete pStopWatch;
+    return true;
 }
 
-bool Host::startStopWatch(int watchID)
+bool Host::adjustStopWatch(const int id, const qint64 milliSeconds)
```

Comment:
Why is there no name overload of this function?

Reply:
As above.